### ISSUE TITLE
LooseDictionary type not good enough

### ISSUE THREAD

**Issue by duclet** [quasarframework/quasar#11309]
**Describe the bug**
Looks like a recent update started using the type `LooseDictionary` for things like the `field` property of a column in QTable. If we already have our own type for what each row would be, we would lose the ability to use that more specific type which provides better type hinting and validation support.

**Expected behavior**
We should be able to use our own type as before.

**Screenshots**
![Screenshot from 2021-11-12 08-45-12](https://user-images.githubusercontent.com/37915/141477280-e6d86ebc-d1de-4667-bc0b-ac03f1833daa.png)

**Platform (please complete the following information):**
Quasar Version:
@quasar/app Version: 2.3.1

**Comment by dsivachenko** [comment]
+1

**Comment by yusufkandemir** [comment]
After some investigation and testing, dropping `LooseDictionary` and using `any` instead will likely be the solution. See this [TS Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIFcDOYD2BbAETjDmQG8BYAKGVuRhxwC5lspQBzau5AIzigsyfOAC8WIdHl7RkAX2oKq1APQq06KFAjhkARXRxMA5GACeABwiZq5q8gAyjTBALAEYYDhACzyALzkyADaoAAmEAAeyKCsYOwgHAC6LHAgfnIA3NSgkLCIKHoAKnC8ADYQAApQOBaY5Nx0CDhlUiCYQo08tDDAEGVhLGycyAA+yAAUEzUA7ixOOC5uHl4+UGYAlAEAfMhpm9k0dHLBSYdKze1gyM2teO0sxaUV1bWYwQDkt22YH0kBIS6DSO3XofQGLGmODmGmw+CIJC2-l2swAdAwcAAaIFybEg4GgnrgwaTWYsDBwwjEOBIlHQ1H8KAMsR4ni46hnaiqdQAZXQHA41kgYX0hmMUFMlmsOXA0HgSH0JXKVRqdQATASmi0fp18Txev0ScNEmNJlCYftaXt0htDmzTudqJdsDdtfdMGrHkqXqqPZ9vu6-gDgkDKHq6AaIaToeSsLgqYidsg0RjWcc07Qw4SwYbIWTYfGETSk2jGczRBn5JW1MgAEoQOBlMp+TB4RtlZAAIjC0PawAineQETw3jYxFWoaB+uJeehVtmyAA1Mg1ZjkDWigALFDNPAWYAVCU96wgD7XGYCEDILBTug1gBC6DAa7A2+T0Ji9QQFQEzetyBwXgACsIA8PZ6mwYgIBFWIIAAN2gMwZm3bQ10wHAYjAD56hACAODKYAOGAZUcTxTkqCAA).

Here is a workaround until this gets further investigation and get implemented:
```ts
field: (row) => JSON.stringify((row as DataRecordData).fieldData)
```

**Comment by yusufkandemir** [comment]
This has been fixed by #12045 and got released on Quasar v2.4.10.

**Comment by yusufkandemir** [comment]
Quasar v2.4.10 also introduced a new type called `QTableColumn` which supports generics. It can be used instead of `QTableProps['columns']` now, like this:
```ts
import { QTableColumn } from 'quasar';

interface User {
  name: string;
}

const columns: QTableColumn<User>[] = [
  {
    name: 'name',
    label: 'Name',
    // 'row' is type of 'User', so no need to explicitly type it
    field: (row) => row.name,
    align: 'left',
  },
  // ...
];
```
